src/name_operations.jl

```diff
@@ -39,9 +39,25 @@ unname(x::AbstractArray) = x
 
 """
     dimnames(A) -> Tuple
+    dimnames(A, d) -> Symbol
 
-Return the names of all the dimensions of the array `A`.
+Return the names of all the dimensions of the array `A`, 
+or just the one for the `d`-th dimension. 
+
+Gives wildcards `:_` if this is not a `NamedDimsArray`.
+And like `size(A, d)`, it allows `d > ndims(A)`.
 """
-dimnames(::Type{<:NamedDimsArray{L}}) where L = L
+dimnames(::Type{<:NamedDimsArray{L}}) where {L} = L
 dimnames(::Type{<:AbstractArray{T, N}}) where {T, N} = ntuple(_->:_, N)
 dimnames(x::T) where T<:AbstractArray = dimnames(T)
+
+function dimnames(AT::Type{<:AbstractArray{T,N}}, d::Integer) where {T,N}
+    if d in 1:N
+        return dimnames(AT)[d]
+    elseif d>N
+        return :_
```

Comment:
What do we want this case?

Reply:
`axes(1:2, 99)` and `size(1:2, 99)` aren't errors, so I thought this should likewise give something. Is there another logical choice? 

Also, if `v = NamedDimsArray(1:2, :a)` the it's nice if `dimnames(v, 2) == dimnames(permutedims(v),1)`.

Reply:
Fair points, lets put that into a comment in the source

Reply:
I added it to the tests, so that it can't break. Hope that's good?